common/persistence/serialization/thrift_mapper.go

```diff
@@ -562,11 +562,35 @@ func taskListPartitionConfigToThrift(info *TaskListPartitionConfig) *sqlblobs.Ta
 	}
 	return &sqlblobs.TaskListPartitionConfig{
 		Version:            &info.Version,
-		NumReadPartitions:  &info.NumReadPartitions,
-		NumWritePartitions: &info.NumWritePartitions,
+		NumReadPartitions:  common.Int32Ptr(int32(len(info.ReadPartitions))),
```

Comment:
This behavior is inconsistent with Cassandra. In Cassandra, the numReadPartitions is the source of truth. Here, it's not, which is confusing.
And it seems to indicate that the persistence tests don't cover these lines.